### ISSUE TITLE
Add more colors to FlxG

### ISSUE THREAD

**Issue by Gama11** [FlixelCommunity/flixel#141]
There are a couple of public static constants storing color codes in `FlxG`. However, they're quite limited, there's only:
1. `RED`
2. `GREEN`
3. `BLUE`
4. `PINK`
5. `WHITE`
6. `BLACK`

They're actually pretty useful once you find out about them - which I didn't for quite some time, partly because `FlxG` is not the most intuitive place to find them. Something like `FlxColor` might be better suited, but then again, a class named like that is already part of the powertools.

Either way, there should be more than just 6 colors. I'd suggest adding add least `GREY`, `YELLOW`, `PURPLE` and a `LIGHT_` - version of every color.

**Comment by IQAndreas** [comment]
> FlxG is not the most intuitive place to find them. Something like FlxColor might be better suited.

I agree, and had a [similar thought a while back](https://github.com/IQAndreas/flixel/commit/98ddd14717d973963562e75ce178e6b4bff8cb88). There are a few color functions in `FlxU` which in that case should also be moved to `FlxColor`.

> I'd suggest adding add least GREY, YELLOW, PURPLE and a LIGHT_ - version of every color.

I'm not sure where the line is between helpful and overkill. I, like you, would prefer a large set of colors, especially if they are all "out of the way", not cluttering up the static constants in the other classes.

This StackOverflow question may also be of use in this topic:
- [Is there an optimum set of colors for 10 players?](http://gamedev.stackexchange.com/questions/46463/is-there-an-optimum-set-of-colors-for-10-players)

**Comment by Gama11** [comment]
> I'm not sure where the line is between helpful and overkill. I, like you, would prefer a large set of colors, especially if they are all "out of the way", not cluttering up the static constants in the other classes.

Yeah, I think it'd be great to have a large pool of colors, just not in `FlxG`, it's already pretty cluttered as it is. Right now, quite a few relatively essential colors are missing. Plus, there's another set of public static constants there, the debugger layouts.

Moving those functions from `FlxU` to `FlxColor` along with a set of color constants seems like a great idea - but I'm wondering, would this cause any compatibility issues with photonstorm's powertools? Like I mentioned before, there's a class with the exact same name there.

**Comment by IQAndreas** [comment]
> Would this cause any compatibility issues with photonstorm's powertools?

Since [PhotonStorm's version](https://github.com/photonstorm/Flixel-Power-Tools/blob/master/Test%20Suite/src/org/flixel/plugin/photonstorm/FlxColor.as) is in the package `org.flixel.plugin.photonstorm` it won't interfere, unless the user accidentally imports the wrong one of the two classes, or wishes to use both Flixel's and PhotonStorm's versions in the same class (in that case, they would need to explicitly call the class with the package name, for example, `org.flixel.plugin.photonstorm.FlxColor.getRandomColor()` which would just be an annoyance, not a bug.)

Since PhotonStorm's class only contains helper functions, perhaps he would be willing to rename the class `FlxColorUtils` in the next release? If the "future release" of Flixel contains major changes, his library may need to be updated anyway to match Flixel's updates.

It doesn't seem like PhotonStorm has many plans to keep updating Flixel Power Tools, but perhaps we can make an "unofficial community version" of his library too. ;)

**Comment by Gama11** [comment]
> Since PhotonStorm's class only contains helper functions, perhaps he would be willing to rename the class FlxColorUtils in the next release?

Well... I'm pretty sure there won't be a new relase. I think he said 1.9 was the final version. He's doesn't seem to be using flash anymore either.

> It doesn't seem like PhotonStorm has many plans to keep updating Flixel Power Tools, but perhaps we can make an "unofficial community version" of his library too. ;)

Haha, yeah... Perhaps some of the more useful features could be integrated into this community build. There might be a few handy helper functions in his `FlxColor` that could be moved over to the new `FlxColor`.

**Comment by Dovyski** [comment]
I like the idea of adding new colors too. The better place I can think of is a new class named `FlxColor`, like you guys suggested. `FlxG` is really cluttered and there is no point adding more to that already saturated class.

About Photonstorm Power Tools, I think we can get some cool ideas from it and merge them into our `FlxColor` class.

**Comment by WingEraser** [comment]
Here is a lot of colors: https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Color.cs

**Comment by IQAndreas** [comment]
I feel this should go into Flixel v2.57 along with FlxColor. What colors should we add?

Here are some suggestions:
- From Flixel: RED, GREEN, BLUE, PINK, WHITE, BLACK
- [From StarCraft 2](http://gamedev.stackexchange.com/a/46471/24598): Red, Blue, Teal, Purple, Orange, Brown, White, Yellow, Green
- [From WarCraft 3](http://gamedev.stackexchange.com/a/46471/24598): Red, Blue, Teal, Purple, Yellow, Orange, Green, Pink, Grey, Light Blue, Dark Green, Brown
- [From MineCraft](http://minecraft.gamepedia.com/Wool): White, Orange, Magenta, Light Blue, Yellow, Lime, Pink, Gray, Light Gray, Cyan, Purple, Blue, Brown, Green, Red, Black
- [HTML color names](http://en.wikipedia.org/wiki/Web_colors#HTML_color_names): White, Silver, Gray, Black, Red, Maroon, Yellow, Olive, Lime, Green, Aqua, Teal, Blue, Navy, Fuchsia, Purple
- [X11 color names](http://en.wikipedia.org/wiki/Web_colors#X11_color_names): Many!
- [From MonoGame](https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Color.cs): Oodles!

_I have to confess, I think I spent more time on that list than I should have; I'm probably making a bigger deal of this than necessary._ :/

**Comment by IQAndreas** [comment]
The MonoGame one is awesome, but might be overkill since I don't even know what 2/3 of those colors are. ;)

16 colors seems like a good balance to me; I'm a big fan of the _MineCraft colors_ (but perhaps I'm just used to working with them all the time). The 16 _HTML named colors_ sound alright, too.

If people want more, we can place all of the X11 color names in a separate class which we can release as a part of Flixel Power Tools, a separate Flixel plugin, or perhaps just a "pure AS3" plugin (all it uses is `uints` anyway).

**Comment by Dovyski** [comment]
I like the MineCraft one more than the HTML group. I also agree with 16 colors, that's a nice number. No need for more.

**Comment by Dovyski** [comment]
For the sake of organization, I think it's a good idea to close this issue and continue the discussion in #209 (FlxColor). All the new colors presented here will be added to `FlxColor` anyway, as @IQAndreas suggested.
